### PR TITLE
Upgrade to esri-geometry-api version 2.2.0

### PR DESCRIPTION
**Pull request by tdcmeehan**
Updating the esri-geometry-api to leverage better support for geometry collections.

### CODE REVIEW

File: presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java [hunk 1]
```diff
@@ -848,10 +848,10 @@ public void testSTUnion()
 
         // within union
         assertUnion("MULTIPOINT ((20 20), (25 25))", "POINT (25 25)", "MULTIPOINT ((20 20), (25 25))");
-        assertUnion("LINESTRING (20 20, 30 30)", "POINT (25 25)", "LINESTRING (20 20, 30 30)");
+        assertUnion("LINESTRING (20 20, 30 30)", "POINT (25 25)", "LINESTRING (20 20, 25 25, 30 30)");
```
Comment (tdcmeehan) [inline presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java]: Note the addition of point (25, 25) on the output.

**Comment by martint** [comment]
What’s the motivation for updating? Are there specific bug fixes or performance improvements that would benefit presto?

**Comment by mbasmanova** [comment]
@martint  2.2.0 version adds support for geometry collections to multiple operations: https://github.com/Esri/geometry-api-java/issues/176 . It will allow us to fix #10628

**Comment by martint** [comment]
Can we add that to the commit message?

**Comment by tdcmeehan** [comment]
My testing has discovered a corner case:

Esri/geometry-api-java#192

It's probably best to wait until that is fixed before we update this dependency.

**Comment by mbasmanova** [comment]
@tdcmeehan Tim, thanks for reporting this issue. It would be nice to fix it, but in the meantime we can workaround by adding an `geometry.isEmpty()` check before calling `estimateMemorySize()`. The latest version of the library didn't include any changes to this API, so it is an existing issue and shouldn't block the upgrade which adds support for geometry collections.

**Pull request merged**